stdlib/public/Darwin/Foundation/IndexPath.swift

```diff
@@ -617,16 +617,17 @@ public struct IndexPath : ReferenceConvertible, Equatable, Hashable, MutableColl
     
     fileprivate init(nsIndexPath: __shared ReferenceType) {
         let count = nsIndexPath.length
-        if count == 0 {
+        switch count {
+        case 0:
             _indexes = []
-        } else if count == 1 {
+        case 1:
             _indexes = .single(nsIndexPath.index(atPosition: 0))
-        } else if count == 2 {
+        case 2:
             _indexes = .pair(nsIndexPath.index(atPosition: 0), nsIndexPath.index(atPosition: 1))
-        } else {
-            var indexes = Array<Int>(repeating: 0, count: count)
-            indexes.withUnsafeMutableBufferPointer { (buffer: inout UnsafeMutableBufferPointer<Int>) -> Void in
-                nsIndexPath.getIndexes(buffer.baseAddress!, range: NSRange(location: 0, length: count))
+        default:
+            let indexes = Array<Int>(unsafeUninitializedCapacity: count) { buf, initializedCount in
```

Comment:
Nit: can you restore the variable name `buffer`? It would cause the next line to be unmodified, which has the benefit of leaving the diff very tidy.

Reply:
done